src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,94 @@
 package core.basesyntax;
 
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
 public class WorkWithFile {
+    private static final String DATA_SEPARATOR = ",";
+    private static final String RESULT = "result";
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        List<String> lines = readFromFile(fromFileName);
+        Map<String, Integer> aggregatedData = createAggregatedData(lines);
+        String report = createReport(aggregatedData);
+        saveToFile(report, toFileName);
+    }
+
+    private String createReport(Map<String, Integer> dataForReport) {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append(createLine(SUPPLY, dataForReport.get(SUPPLY)));
+        stringBuilder.append(createLine(BUY, dataForReport.get(BUY)));
+        stringBuilder.append(createLine(RESULT, dataForReport.get(RESULT)));
+        return stringBuilder.toString();
+    }
+
+    private String createLine(String key, Integer value) {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append(key)
+                .append(DATA_SEPARATOR)
+                .append(value)
+                .append(System.lineSeparator());
+        return stringBuilder.toString();
+    }
+
+    private void saveToFile(String report, String toFileName) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(toFileName))) {
+            writer.write(report);
+        } catch (IOException e) {
+            throw new RuntimeException("can't write to the file", e);
```

This violates checklist item "Don't ignore exceptions": the writer catch rethrows a RuntimeException with the vague message "can't write to the file". Include the target filename and keep the original exception as the cause for better diagnostics. Example fix: `throw new RuntimeException("Can't write data to the file " + toFileName, e);`